Makefile

```
@@ -26,10 +26,10 @@ build: $(EMBEDDED_MANIFESTS_TARGET)
 install:
 	go install cmd/flux
 
+install-dev: $(EMBEDDED_MANIFESTS_TARGET)
+	CGO_ENABLED=0 go build -o /usr/local/bin ./cmd/flux
```

This has never worked on my system, as `/usr/local/bin` is protected and I do not run my `make` commands with `sudo`:

```console
$ make install-dev
CGO_ENABLED=0 go build -o /usr/local/bin ./cmd/flux
go build github.com/fluxcd/flux2/cmd/flux: copying /tmp/go-build3338692359/b001/exe/a.out: open /usr/local/bin/flux: permission denied
make: *** [Makefile:35: install-dev] Error 1
```

If you change it to something like this, it will prompt for a password, while (hopefully) not creating any problems on MacOS:

```suggestion
install-dev: $(EMBEDDED_MANIFESTS_TARGET)
	CGO_ENABLED=0 go build -o /tmp/flux-dev ./cmd/flux
	sudo install -Dm755 /tmp/flux-dev /usr/local/bin/flux
```

---

Seems that we are in a deadlock here, you don't want sudo make and I don't want make sudo.

---

`sudo make` would make Go run as root, which is not something you want as all Go Modules are cached for the normal user.

My suggestion makes this target work on any system, and the issue you have with it for MacOS can be overcome by running `brew` with the `--overwrite` flag. Given all this, I see much stronger arguments in favor of my suggestion, than against.

---

I've undone the changes to Makefile